### PR TITLE
WIP:Classification:RF Model

### PR DESCRIPTION
**Pull request by lichenzhi**
This script contains two functions. The first one is the Random Forest model with cross validation. And the second one is to create the summary for precision, recall and F-1 measure. The training result is pretty good, almost around 100%. However, for the testing result, it's only about 30%.

### CODE REVIEW

**Comment by lichenzhi** [comment]
@lizhua @ye-zhi

**Comment by lizhua** [comment]
The problem is the data loading function. We need to use load_mouseday_features() to load sufficient features for classification.

**Comment by jarrodmillman** [comment]
You need to write tests for your functions.  Also you should use lower case names for Python modules (i.e., files).

File: mousestyles/classification/Classification.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+from mousestyles import data
+import numpy as np
+from sklearn.grid_search import GridSearchCV
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.preprocessing import StandardScaler
+import pandas as pd 
+
+def RandomForest():
+	df = data.load_all_features()
```
Comment (ye-zhi) [inline mousestyles/classification/Classification.py]: We should not load the data inside the function. We need to take data as an input so user can define their own data. @lichenzhi @lizhua @changsiyao

File: mousestyles/classification/Classification.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	#get the recall list 
+	recall = [prediction_accurate_rate_matrix.T.iloc[i,i] for i in range(prediction_accurate_rate_matrix.shape[1]-1)]
+	#get the F1 list 
+	f1= [2*precision[i]*recall[i]/(precision[i]+recall[i]) for i in range(prediction_accurate_rate_matrix.shape[1]-1)]
+	summary = pd.concat([pd.DataFrame(precision),pd.DataFrame(recall),pd.DataFrame(f1)],axis=1)
+	summary.columns = ['precision','recall',"F1_score"]
+	return(summary)
+
+#test with RF Model 
+GetSummary(RandomForest())
```
Comment (ye-zhi) [inline mousestyles/classification/Classification.py]: should delete the last line. This script should include the functions only @lichenzhi

**Comment by ye-zhi** [comment]
@lichenzhi @lizhua @changsiyao  I have added some line comments, please fix them.

**Comment by changsiyao** [comment]
So one of you, Ye or Zhuangdi needs to write tests, which preferable can be included in this pull request as well, so the coverage won't drop.
That being said, you should either all push into my fork, or Chenzhi need to set up her fork and add Ye and Zhuangdi as collaborators and you all push to Chenzhi's fork.

File: mousestyles/classification/Classification.py (hunk 1 above)
Comment (changsiyao) [inline mousestyles/classification/Classification.py]: Actually, consider taking train_X, train_Y, test_X, test_Y as inputs, this way you don't have to split data into labels and features inside the function. It avoid mistakes that originate from unexpected data.

File: mousestyles/classification/Classification.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+from mousestyles import data
+import numpy as np
+from sklearn.grid_search import GridSearchCV
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.preprocessing import StandardScaler
+import pandas as pd 
+
+def RandomForest():
```
Comment (changsiyao) [inline mousestyles/classification/Classification.py]: Here, you need to add doc string for the function, specify what the function does, what parameters it take, what output it returns, provide examples etc

File: mousestyles/classification/Classification.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	#eliminate columns strain, mouse, day, hour
+	train_x = train.iloc[:,4:]
+	#label is the strain 
+	train_y = train['strain']
+	test_x = test.iloc[:,4:]
+	test_y = test['strain']
+	#creat RF model 
+	scaler = StandardScaler()
+	train_x = scaler.fit_transform(train_x)
+	es = [500,1000]
```
Comment (changsiyao) [inline mousestyles/classification/Classification.py]: @lichenzhi @lizhua I think these grid search parameters should be defined by users as well, so maybe take them in as inputs.

File: mousestyles/classification/Classification.py (hunk 3 above)
Comment (pttzty) [inline mousestyles/classification/Classification.py]: Here I strongly suggest making the function RandomForest() to take a numpy array or pandas dataframe as the input, so that the user could specify whether to use a subset the data or use all data rows to fit the random forest. The function should also take the parameters like the number of trees, the number of features used in each split, and the tree size to be the input parameters so that it would be easier in future parameter tuning.

File: mousestyles/classification/Classification.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	prediction = clf.predict(scaler.fit_transform((test_x)))
+	prediction = pd.DataFrame(prediction)
+	#reindex test_y so that it starts from 0  
+	test_y.index = range(test_y.shape[0])
+	predict_true = pd.concat([prediction,test_y],axis=1)
+	predict_true.columns = ['predict_strain','true_strain']
+	return(predict_true)
+
+
+
```
Comment (z357412526) [inline mousestyles/classification/Classification.py]: I recommend you to change the functions' name so that they are consistent with code in other files.

**Comment by lichenzhi** [comment]
__
